test/interpreter_scripts/in/namespace.txt

```diff
@@ -0,0 +1,15 @@
+namespace NAMESPACE {
+    test = "TEST"
+    //assert( NAMESPACE.test = "TEST" ) //DO THIS
+    
+    namespace inner {
+        test = "inner"
+    }
+}
+
+test = "OUTER"
+assert(test = "OUTER")
+assert( NAMESPACE.test = "TEST" )
+//assert( NAMESPACE.inner.test = "inner" ) //DO THIS
```

Comment:
When you allow types involving scope (namespace, struct, class, etcetera), then the symbol resolution and type system are coupled.

How coupled? I think symbol declaration and normal references can still be in an initial pass. Accessors will need to be resolved with types, but resolving reads shouldn't be so challenging.